### ISSUE TITLE
[Epic] Eventonica v2021

### ISSUE THREAD

**Issue by vegetabill** [Techtonica/curriculum#1296]
## TL;DR
High-level improvements to Eventonica based on H1-2020 cohort's experience along with Alina & Bill's Opinions.

## Background
In cohorts so far, the main project used to learn full-stack app development is [Eventonica](), a CRUD app for handling events of different categories. It was divided into [Parts 1-7](https://github.com/Techtonica/curriculum/blob/master/projects/eventonica/README.md).

## Eventonica Tenets
- Use the same reqs over and over to make it about the tech/technique, not the domain
- The more milestones the better, at least weekly
- Scheduling should occur such that Eventonica is reinforcing topics they have already covered at least once, although maybe not all combined in one task

## New Sequence

[Intro readme](https://github.com/Techtonica/curriculum/blob/main/projects/eventonica/README.md) should list user story-style features and these will be used throughout all the assignments.

| Topic                       	| H1-2020 Sequence 	| H1-2021 Sequence 🔼	| Issue(s)              	| Notes           	|
|-----------------------------	|:-----------------:	|:------------:	|-----------------------	|-----------------	|
| Models and Behavior         	|       Part 1      	|    Part 1    	|  #1420 ✅ 	|                 	|
| Jasmine Unit Testing                     	|       Part 2      	| Part 1.1 	|     -                  	|  	|
| Handwritten Web (HTML/CSS/JS DOM) 	|       Part 3      	|    Part 2    	|    #1419  ✅                 	|                 	|
| External Ticketmaster API   	|       Part 4      	|  - 	|  #1424                     	|  Feature cut               	|
| Express API (no DB)     	|       Part 5      	|    Part 3    	|   #1425                    	|                 	|
|  API Tests     	|       -      	|    Part 3.1    	|   #1215 WIP                    	|  Not a must-have. This is new and not essential               	|
| Backend uses Postgres                    	|       Part 6      	|    Part 4    	|   #1427                    	|                 	|
| React (client state only)           	|       Part 7      	|    Part 5    	|    #1421                   	|    Prereq: new React Part 4, #1337             	|
| Enzyme Testing           	|       -      	|    Part 5.1    	|   TBD - cut from 2021 scope                    	|     Practice writing Enzyme tests for one component      	|
| React calls API             	|         -         	|    Part 6    	|    #1428                 |  This was a really challenging and complex transition from React with client state only so we need some dedicated lessons/activities on this to ensure adequate prep. |

#### Optional Items
- #1429 Reinforcement Lesson for splitting/refactoring a frontend and backend (nice-to-have)

### Child Issues
- Eventonica README should contain master requirements dock with mocks to be used for entire sequence of deliverables - it should be a very slim MVP with CRUD events and CRUD users. The one join data is users have a set of favorited events. All other features currently in the parts should be either cut (Ticketmaster) or moved into challenges.
- Using above req doc, convert the existing jQuery lesson (part 1) into plain JS DOM functions

For FAQs on how to contribute, see: https://github.com/Techtonica/curriculum/blob/main/CONTRIBUTING.md

**Comment by alodahl** [comment]
Opinions:

I think we can limit use of jQuery, but I don't think it's throwaway.  It's legacy, but a lot of companies still use it, and I think it's a good way to introduce methods and use of JS in the dom.  I can also be convinced to just teach vanilla JS for this purpose, though. But by the logic of the first comment, we could just teach them React instead of HTML too.

I also think some of this kinda cuts into the core of project-based learning, so I think we do need to have a phone call about it. I'll ping you.

**Comment by alodahl** [comment]
Notes from 04/27 Bill:Alina 1-on-1:

- between part 4 and part 5, either there is not enough direction on how to build servers, or the instructions are too vague in Part 5. Let's look over those parts and decide what to change.
- Cut outlines into smaller parts. Maybe two assessments/checkins per week once you get to Parts 4, 5, 6
- Remove extra methods in early eventonica js challenge - we don't need a bunch of features.

**Comment by vegetabill** [comment]
_Archived feedback notes from Eventonica experience_


## Issues / Feedback with Existing Project (Eventonica)
- Learning jQuery (Part 3) was mainly throwaway since we ultimately want to learn React
- Part 5 was a massive jump in complexity that needed a lot more hand-holding and background in distributed app architecture
- Testing is not really feasible whilst learning something so Part 2 was not worth the time
- Back-loading React made it one of the most difficult parts of the current curriculum
- The assignment was divided into 7 parts but actually each part was still too large as a checkpoint to turn in for grading
- Overall length was too long (curriculum length was over 1 month)
  - We want to strike a balance. Working on the same project gets old - also if anyone falls behind they can't just skip and catch up easily

**Comment by alodahl** [comment]
jQuery instances are listed here: https://github.com/Techtonica/curriculum/pull/1371

**Comment by vegetabill** [comment]
@alodahl I added a table to visualize the parts of this epic above and I'll try to further break it down over the next week.

**Comment by alodahl** [comment]
Does no issue mean that it's done, or that it needs an issue made?

**Comment by vegetabill** [comment]
@alodahl I've added placeholder tickets so now there is either an issue or an explicit "none" for each and I think this now represents the whole scope.

**Comment by vegetabill** [comment]
All items are merged or closed. Closing this epic.

Future ideas can be added to => #1542
